cmd/solo-kit-gen/main.go

```diff
@@ -72,11 +63,31 @@ func run() error {
 
 		var descriptors []*descriptor.FileDescriptorSet
 
-		if err := filepath.Walk(inDir, func(path string, info os.FileInfo, err error) error {
-			if !strings.HasSuffix(path, ".proto") {
+		var projectGoPackage string
+		if err := filepath.Walk(inDir, func(protoFile string, info os.FileInfo, err error) error {
```

Comment:
This is a 30 line nested lambda, I think we should factor it out -- pretty confusing to figure out what this is doing.